### PR TITLE
FIX use sparse_output parameter for OneHotEncoder

### PR DESCRIPTION
**Pull request by ogrisel**
`OneHotEncoder.sparse` is deprecated in favor of `OneHotEncoder.sparse_output`

### CODE REVIEW

File: python_scripts/03_categorical_pipeline_ex_02.py [hunk 1]
```diff
@@ -102,7 +102,7 @@
 #
 # Hint: `HistGradientBoostingClassifier` does not yet support sparse input
 # data. You might want to use
-# `OneHotEncoder(handle_unknown="ignore", sparse=False)` to force the use of a
+# `OneHotEncoder(handle_unknown="ignore", sparse_output=output=False)` to force the use of a
```
Comment (ArturoAmorQ) [inline python_scripts/03_categorical_pipeline_ex_02.py:105]:
```suggestion
# `OneHotEncoder(handle_unknown="ignore", sparse_output=False)` to force the use of a
```

File: python_scripts/03_categorical_pipeline_sol_02.py [hunk 2]
```diff
@@ -126,7 +126,7 @@
 #
 # Hint: `HistGradientBoostingClassifier` does not yet support sparse input
 # data. You might want to use
-# `OneHotEncoder(handle_unknown="ignore", sparse=False)` to force the use of a
+# `OneHotEncoder(handle_unknown="ignore", sparse_output=output=False)` to force the use of a
```
Comment (ArturoAmorQ) [inline python_scripts/03_categorical_pipeline_sol_02.py:129]:
```suggestion
# `OneHotEncoder(handle_unknown="ignore", sparse_output=False)` to force the use of a
```

File: python_scripts/linear_models_sol_04.py [hunk 3]
```diff
@@ -208,7 +208,7 @@
 from sklearn.preprocessing import OneHotEncoder
 
 single_feature = ["CentralAir"]
-encoder = OneHotEncoder(sparse=False, dtype=np.int32)
+encoder = OneHotEncoder(sparse_output=output=False, dtype=np.int32)
```
Comment (ArturoAmorQ) [inline python_scripts/linear_models_sol_04.py:211]:
```suggestion
encoder = OneHotEncoder(sparse_output=False, dtype=np.int32)
```

File: python_scripts/03_categorical_pipeline.py [hunk 4]
```diff
@@ -172,13 +172,13 @@
 # %%
 from sklearn.preprocessing import OneHotEncoder
 
-encoder = OneHotEncoder(sparse=False)
+encoder = OneHotEncoder(sparse_output=output=False)
```
Comment (ArturoAmorQ) [inline python_scripts/03_categorical_pipeline.py:175]:
```suggestion
encoder = OneHotEncoder(sparse_output=False)
```

**Comment by ArturoAmorQ** [comment]
I took the liberty to fix a typo :)

**Review by ArturoAmorQ — APPROVED** [review]

**Comment by ogrisel** [comment]
Thanks for the fix, and sorry for opening the PR without reviewing the diff...

**Pull request merged**